queries/latex/highlights.scm

```diff
@@ -53,12 +53,14 @@
 (begin
   command: _ @module
   name: (curly_group_text
-    (text) @label @nospell))
+    (text) @label @nospell)
+  (#not-has-ancestor? @label math_environment))
```

Comment:
`has-ancestor` is relatively expensive, so try to avoid that at all cost. It's better to rely on query ordering to get the desired final result. (It's perfectly fine to have multiple captures apply, as long as the right one "wins", i.e., is last.)

Reply:
Thanks for the helpful insight. However the order of the queries seems to be corrected, i.e. the math_environment is second to the generic environment, but it is not the one to "win".

Reply:
Yes, because that only covers the environment itself. You need to override the `name` capture there. But I'm not sure this is wanted -- there's been some back and forth over having the math environment names be the same as the contents or the same as the other environments.